bin/commands/cmd_gen.ml

```diff
@@ -30,7 +30,35 @@ let rec atdgen ?(rescript = false) ?output_prefix:original_prefix = function
     | 0, i | i, _ ->
       Error (`Atd_error (i, file)))
 
-let run ?output_prefix ?(rescript = false) files =
+let rec checkout ~git_file = function
+  | [] ->
+    Ok []
+  | file :: tail ->
+    let open Letops.Result in
+    let* version, commit = Generator.split_commit file in
+    let* filename, _extension = Generator.split_filename git_file in
+    (* This checks out the commit of the versioned file in the same location as the file.
+     * Dangerous?! Should it be in output_prefix, or in a temporary directory?
+     *)
+    let temp_file = filename ^ "_" ^ string_of_int version ^ ".atd" in
+    let commit = match commit with None -> "HEAD" | Some commit -> commit in
```

Comment:
Good thinking! The `None` case is a version number without commit hash as in Wojtek's example in #12.
Is it possible that a version without commit hash is ever _not_ the last argument? Like, a reverse migration, is that a valid use case?
Otherwise we should probably generate an error if `tail` is not empty and the commit hash is `None`.

Btw, I just realized we can use `Option.value` here for clarity.

Reply:
Do we want HEAD or do we want whatever's in the current working directory?

Reply:
Yes, maybe it should not be HEAD. What's the typical workflow here, do users run stork before committing their files?

Reply:
I think yes

Reply:
Okay, I changed that behavior. No commit hash -> take what's in the working directory.